src/components/Reports/AdditionalSalaryRequest/Shared/AdditionalSalaryRequestContext.tsx

```diff
@@ -40,70 +46,63 @@ export const AdditionalSalaryRequestProvider: React.FC<Props> = ({
   children,
 }) => {
   const steps = useStepList(FormEnum.AdditionalSalary);
-  const totalSteps = steps.length;
+  const objects = useMemo(
+    () => Object.values(AdditionalSalaryRequestSectionEnum),
+    [],
+  );
 
   // Step Handlers
   const [currentStep, setCurrentStep] = useState(
     AdditionalSalaryRequestSectionEnum.AboutForm,
   );
   const [percentComplete, setPercentComplete] = useState(33);
-
   const [currentIndex, setCurrentIndex] = useState(0);
 
-  const handleNextStep = () => {
-    setCurrentStep((prevStep) => {
-      const next =
-        prevStep === AdditionalSalaryRequestSectionEnum.AboutForm
-          ? AdditionalSalaryRequestSectionEnum.CompleteForm
-          : prevStep === AdditionalSalaryRequestSectionEnum.CompleteForm
-            ? AdditionalSalaryRequestSectionEnum.Receipt
-            : prevStep;
-
-      const newIndex = currentIndex + 1;
-      handleNextIndexChange(newIndex);
-      handlePercentComplete(newIndex);
-      return next;
-    });
-  };
+  useEffect(() => {
+    setPercentComplete(
+      ((objects.indexOf(currentStep) + 1) / objects.length) * 100,
+    );
+  }, [currentStep]);
```

Comment:
Since the completion percentage can always be derived from the current step, I'd recommend this. Here's an article that goes into more detail on why this is preferred to useState+useEffect: https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

```suggestion
  const percentComplete = ((objects.indexOf(currentStep) + 1) / objects.length) * 100;
```